capsules/src/usb/usbc_client_ctrl.rs

```diff
@@ -22,58 +34,60 @@ use core::cmp::min;
 use kernel::hil;
 use kernel::hil::usb::TransferType;
 
-const DESCRIPTOR_BUFLEN: usize = 64;
+const DESCRIPTOR_BUFLEN: usize = 128;
 
 const N_ENDPOINTS: usize = 3;
 
-pub struct ClientCtrl<'a, 'b, C: 'a> {
-    // The hardware controller
-    controller: &'a C,
+/// Handler for USB control endpoint requests.
+pub struct ClientCtrl<'a, 'b, U: 'a> {
+    /// The USB hardware controller.
+    controller: &'a U,
 
-    // State for tracking each endpoint
+    /// State of each endpoint.
     state: [Cell<State>; N_ENDPOINTS],
 
-    // A 64-byte buffer for the control endpoint
+    /// A 64-byte buffer for the control endpoint to be passed to the USB
+    /// driver.
     ctrl_buffer: Buffer64,
 
-    // Storage for composing responses to device-descriptor requests
+    /// Storage for composing responses to device descriptor requests.
     descriptor_storage: [Cell<u8>; DESCRIPTOR_BUFLEN],
 
-    // Descriptors to reply to control requests
-    device_descriptor: DeviceDescriptor,
-
-    // For now we only support one configuration...
-    configuration_descriptor: ConfigurationDescriptor,
+    /// Buffer containing the byte-packed representation of the device
+    /// descriptor. This is expected to be created and passed from the user of
+    /// `ClientCtrl`.
+    device_descriptor_buffer: DeviceBuffer,
 
-    // ...with only one interface
-    interface_descriptor: InterfaceDescriptor,
+    /// Buffer containing the byte-packed representation of the configuration
+    /// descriptor and all other descriptors for this device.
+    other_descriptor_buffer: DescriptorBuffer,
 
-    // A list of endpoints for the configuration
-    endpoint_descriptors: &'b [EndpointDescriptor],
-
-    // A HID descriptor for the configuration, if any
+    /// An optional HID descriptor for the configuration. This can be requested
+    /// separately. It should also be included in `other_descriptor_buffer`.
     hid_descriptor: Option<&'b HIDDescriptor<'b>>,
 
-    // A report descriptor for the configuration, if any
+    /// An optional report descriptor for the configuration. This can be
+    /// requested separately. It should also be included in
+    /// `other_descriptor_buffer`.
```

Comment:
```suggestion
    /// An optional report descriptor for the configuration. This can be
    /// requested separately. It must also be included in
    /// `other_descriptor_buffer` if it exists.
```

Reply:
This is the other location that same language is used and is still relevant.